specs/src/tx_client.md

```diff
@@ -0,0 +1,273 @@
+
+**Important**: These are the only reliable ways to submit transactions. Other patterns (like multiple transactions from one account without confirmation) will cause sequence mismatches and failures.
+
+### Usage Modes
+
+1. **Ordered Submission:** (Low Throughput, Sequential Processing, Ordering guarantees):
+    - **Single Account**: Uses one account for all transactions
+    - **Sequential Processing**: One transaction per block, confirmed before next submission
+    - **Ordering Guarantees**: Transactions are processed in exact submission order
+    - **Use Case**: Applications requiring strict transaction ordering
+
+2. **Unordered Submission:** (High Throughput, Parallel Processing, No Ordering)
+    - **Worker Accounts**: Master account creates multiple worker accounts with fee grants
+    - **Parallel Processing**: Multiple transactions submitted concurrently across workers
+    - **No Ordering Guarantees**: Transactions will most likely not be ordered.
+    - **Use Case**: Applications prioritizing throughput over ordering
+
+## Protocol/Component Description
+
+### Signer
+
+The Signer is a core component that handles transaction creation, signing, and encoding. It manages multiple accounts and provides the cryptographic operations needed for transaction submission.
+
+**Key Functionality**:
+
+- **Transaction Building**: Creates transactions from messages and blobs
+  - `CreateTx(msgs []sdktypes.Msg, opts ...TxOption) ([]byte, authsigning.Tx, error)`
+  - `CreatePayForBlobs(accountName string, blobs []*share.Blob, opts ...TxOption) ([]byte, uint64, error)`
+- **Signing**: Signs transactions using account private keys
+  - `SignTx(msgs []sdktypes.Msg, opts ...TxOption) (authsigning.Tx, string, uint64, error)`
+- **Encoding/Decoding**: Converts transactions to bytes and vice versa
+  - `EncodeTx(tx sdktypes.Tx) ([]byte, error)`
+  - `DecodeTx(txBytes []byte) (authsigning.Tx, error)`
+- **Account Management**: Tracks account sequences and manages multiple signers
+  - `IncrementSequence(accountName string) error`
+  - `SetSequence(accountName string, seq uint64) error`
+
+**How TxClient Uses Signer**: The TxClient delegates all transaction creation, signing, and encoding operations to the Signer, which handles sequence management, multi-account support, and blob transaction processing.
+
+### Worker Queue / Parallel Submission
+
+The worker queue is critical for unordered submission mode, enabling concurrent transaction processing without sequence contention.
+
+**Key Functionality**:
+
+- **Job Queue**: Holds submission jobs (`SubmissionJob`) containing blobs, options, and response channels
+- **Workers**:
+  - Worker 0 = primary account
+  - Workers >0 = additional accounts created and fee-granted by primary
+- **Parallelism**: Allows concurrent submission of multiple transactions without sequence contention
+- **Fee Management**: Non-primary workers use fee grants so all transaction fees are paid by the primary account
```

Comment:
Could also mention that the workers get unlimited feegrant from master account.

Reply:
and that these accounts are reused between runtimes (if parallel tx submission is enabled)